relay-spans/src/lib.rs

```diff
@@ -0,0 +1,247 @@
+pub struct Link {
+    pub trace_id: String,
+    pub span_id: String,
+    pub trace_state: String,
+    pub attributes: Vec<KeyValue>,
+    pub dropped_attributes_count: u32,
+}
+
+#[derive(Debug, Deserialize_repr)]
+#[repr(u32)]
+pub enum SpanKind {
+    Unspecified = 0,
+    Internal = 1,
+    Server = 2,
+    Client = 3,
+    Producer = 4,
+    Consumer = 5,
+}
+
+impl SpanKind {
+    pub fn as_str_name(&self) -> &'static str {
+        match self {
+            SpanKind::Unspecified => "SPAN_KIND_UNSPECIFIED",
+            SpanKind::Internal => "SPAN_KIND_INTERNAL",
+            SpanKind::Server => "SPAN_KIND_SERVER",
+            SpanKind::Client => "SPAN_KIND_CLIENT",
+            SpanKind::Producer => "SPAN_KIND_PRODUCER",
+            SpanKind::Consumer => "SPAN_KIND_CONSUMER",
+        }
+    }
+    pub fn from_str_name(value: &str) -> ::core::option::Option<Self> {
+        match value {
+            "SPAN_KIND_UNSPECIFIED" => Some(Self::Unspecified),
+            "SPAN_KIND_INTERNAL" => Some(Self::Internal),
+            "SPAN_KIND_SERVER" => Some(Self::Server),
+            "SPAN_KIND_CLIENT" => Some(Self::Client),
+            "SPAN_KIND_PRODUCER" => Some(Self::Producer),
+            "SPAN_KIND_CONSUMER" => Some(Self::Consumer),
+            _ => None,
+        }
+    }
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Status {
+    pub message: Option<String>,
+    pub code: StatusCode,
+}
+
+#[derive(Debug, Deserialize_repr)]
+#[repr(u32)]
+pub enum StatusCode {
+    Unset = 0,
+    Ok = 1,
+    Error = 2,
+}
+
+impl StatusCode {
+    pub fn as_str_name(&self) -> &'static str {
+        match self {
+            StatusCode::Unset => "STATUS_CODE_UNSET",
+            StatusCode::Ok => "STATUS_CODE_OK",
+            StatusCode::Error => "STATUS_CODE_ERROR",
+        }
+    }
+    pub fn from_str_name(value: &str) -> ::core::option::Option<Self> {
+        match value {
+            "STATUS_CODE_UNSET" => Some(Self::Unset),
+            "STATUS_CODE_OK" => Some(Self::Ok),
+            "STATUS_CODE_ERROR" => Some(Self::Error),
+            _ => None,
+        }
+    }
+}
+
+#[derive(Debug, Deserialize)]
+pub struct AnyValue {
+    pub value: Option<Value>,
+}
+
+#[derive(Debug, Deserialize)]
+pub enum Value {
+    StringValue(String),
+    BoolValue(bool),
+    IntValue(i64),
+    DoubleValue(f64),
+    ArrayValue(ArrayValue),
+    KvlistValue(KeyValueList),
+    BytesValue(Vec<u8>),
+}
+
+#[derive(Debug, Deserialize)]
+pub struct ArrayValue {
+    pub values: Vec<AnyValue>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct KeyValueList {
+    pub values: Vec<KeyValue>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct KeyValue {
+    pub key: String,
+    pub value: Option<AnyValue>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct InstrumentationScope {
+    pub name: String,
+    pub version: String,
+    pub attributes: Vec<KeyValue>,
+    pub dropped_attributes_count: u32,
+}
+
+impl From<Span> for EventSpan {
```

Comment:
@phacops we briefly discussed this offline: I'm wondering if instead of converting to `EventSpan`, we should convert event spans to `OtelSpan` instead when we extract them from transactions?

I realize that this would slow down development because we'd have to make sure that the sentry consumer supports the OTEL format first, so I am also fine with converting to event spans for now.

Reply:
Yes, that's precisely why I started like this. I realize in the long term, we want the reverse but we're long from being ready for this.

I think in the interest of figuring out how an OTel SDK would interact with this, it's still worth doing this and then reversing when we're ready.